src/Http/WebUtilities/src/MultipartSectionPipeReader.cs

```diff
@@ -0,0 +1,362 @@
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.WebUtilities
+{
+    public class MultipartSectionPipeReader : PipeReader
+    {
+        private readonly PipeReader _pipeReader;
+        private bool _finished = false;
+        private bool _metadataSkipped = false;
+        private readonly MultipartBoundary _boundary;
+        private int _partialMatchIndex = 0;
+        private ReadOnlySequence<byte> _buffer;
+        private SequencePosition _endPosition;
+        private bool _isReaderComplete;
+
+        private static ReadOnlySpan<byte> CrlfDelimiter => new byte[] { (byte)'\r', (byte)'\n' };
+        private static ReadOnlySpan<byte> EndOfFileDelimiter => new byte[] { (byte)'-', (byte)'-' };
+
+        public bool CanSeek { get; }
```

Comment:
Is this still used anywhere? I don't see any callers.

Reply:
They are called in TrySkipMetaData

Reply:
Not anymore?

Reply:
I am now realizing you meant CanSeek.